add_definitions parses the flags and adds wrong quotes if the the
first definition argument has a value.

-DFOO=1 -DBAR turns into -DFOO="1 -BAR"

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,7 +167,7 @@
 
 message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}: ${LLVM_CMAKE_DIR}")
 include_directories(SYSTEM ${LLVM_INCLUDE_DIRS})
-add_definitions(${LLVM_DEFINITIONS})
+add_compile_options(${LLVM_DEFINITIONS})
 
 string(REGEX REPLACE "(-rc[0-9]+|git)" "" CLANG_VERSION ${LLVM_PACKAGE_VERSION})
 
